code/controllers/subsystem/processing/inbounds.dm

```diff
@@ -0,0 +1,35 @@
+var/datum/subsystem/objects/SSinbounds
+/datum/subsystem/inbounds
+	name = "Inbounds"
+	priority = 40
+	flags = SS_NO_INIT
+
+	var/list/processing = list()
+	var/list/currentrun = list()
+
+/datum/subsystem/inbounds/New()
+	NEW_SS_GLOBAL(SSinbounds)
+/datum/subsystem/inbounds/stat_entry()
+	..("P:[processing.len]")
+
+/datum/subsystem/inbounds/fire(resumed = 0)
+	if (!resumed)
+		src.currentrun = processing.Copy()
+	//cache for sanic speed (lists are references anyways)
+	var/list/currentrun = src.currentrun
+
+	while(currentrun.len)
+		var/atom/thing = currentrun[currentrun.len]
+		currentrun.len--
+		if(thing)
+			thing.check_in_bounds(wait)
+		else
+			SSobj.processing -= thing
```

Comment:
Yes. Because I need the processing subsystem,  but for it to call a different proc.